Various fixes

carlopi opened the pull request.

carlopi committed:
Fix warnings: move -> std::move

carlopi committed:
Makefile: python -> python3

carlopi committed:
Fix to https://github.com/duckdb/duckdb/discussions/5461

carlopi committed:
valgrid: Initialize version for ColumnScanState

This silences many valgrind flaky results

carlopi committed:
CI: Fix failure detection for Regression.yml

Idea is tests are executed in any case, one after the other (even with
previous failures), but still failures have to be reported.
This fixes logic added in 0ba4d25015014f856ea991033eb9d1fd891b4317.

carlopi committed:
third_party/fmt: Use char8_t

carlopi committed:
Fix warnings: (void)x syntax to mark x as used

carlopi commented:
This should fix the warnings detected by CRAN + other minor fixes.

On https://github.com/duckdb/duckdb/pull/6036/commits/4e595b1c0ddefa1cdc9c609b5745b6cd50e317c9, adding the initialization should do no harm, but valgrid signals that sometimes on current master it gets to the D_ASSERT(state.version == version) with uninitialized data.
```
idx_t ColumnData::ScanVector(ColumnScanState &state, Vector &result, idx_t remaining) {
        state.previous_states.clear();
        if (state.version != version) {
                InitializeScanWithOffset(state, state.row_index);
                state.current->InitializeScan(state);
                state.initialized = true;
        } else if (!state.initialized) {
                D_ASSERT(state.current);
                state.current->InitializeScan(state);
                state.internal_index = state.current->start;
                state.initialized = true;
        }
        D_ASSERT(data.HasSegment(state.current));
        D_ASSERT(state.version == version);
        // more stuff
```
@Mytherin: Should this not be inverted, with the check on initialization being done first, and only comparing version between initialized values?

Mause commented on .github/workflows/Regression.yml:
```diff
@@ -66,31 +66,31 @@ jobs:
 
     - name: Regression Test TPCH
-      continue-on-error: true
+      if: ${{ always() }}
```
Nitpick, but these can be simplified to `if: always()` 😉

carlopi replied:
Cleaned up, thanks!

carlopi committed:
Fix syntax in Regression.yml (NFC)

This simplifies syntax introduced earlier in 61e7daa74055b3b6c455ad740fb77ecc7697f748

Tmonster commented on third_party/fmt/include/fmt/core.h:
```diff
@@ -356,7 +356,11 @@ using string_view = basic_string_view<char>;
 using wstring_view = basic_string_view<wchar_t>;
 
 // A UTF-8 code unit type.
+#if FMT_HAS_FEATURE(__cpp_char8_t)
```
haha, yes. Much more eloquent than what I had

Tmonster reviewed (approved):
lgtm!

carlopi committed:
Fix 130e0a11 by patching in generate_grammar.py

carlopi commented on third_party/libpg_query/src_backend_parser_gram.cpp:
```diff
@@ -17298,6 +17298,7 @@ YYLTYPE yylloc;
   yystate = 0;
   yyerrstatus = 0;
   yynerrs = 0;
+  (void)yynerrs;
```
Thanks for the catch, should be addressed here: b20dc943d1a1cbd2c3e0d1f77b6ce3ceb00a6051 (commit on top).

carlopi committed:
node: fix -Wstring-plus-int

carlopi committed:
Merge branch 'master' into variousFixes
